[Travis] quiet catkin build output.

130s opened the pull request:
**Problem**: With a larger repository the job output can easily reach 4MB, which is the current limit of Travis CI that makes hard to debug upon errors.

**Approach**: Add an option to suppress standard output from `caking build`, which often consumes the majority of the lines in the output with many packages built.

<s>**TODO**: Code needs to be optimized (I rather see how it works first before polishing code. Not a good excuse!).</s>

130s commented:
Ready for the review. @ipa-mdl @gavanderhoorn if either/both of you are interested, please.

- Example of successful build in quiet mode ([in industrial_ci](https://travis-ci.org/ros-industrial/industrial_ci/jobs/130141009#L2224))
 ```
A command 'catkin run_tests -i -v --no-deps --no-status industrial_ci -p4 --make-args -j8 --' is running quiet mode. Elapsed: 0
+++catkin run_tests -i -v --no-deps --no-status industrial_ci -p4 --make-args -j8 --
+++trap - EXIT
+++catkin_test_results build
Summary: 0 tests, 0 errors, 0 failures
+++travis_time_end
```

- Example of failing build in quiet mode (in [130s/moveit](https://travis-ci.org/130s/moveit/jobs/129664124#L4150)). As you see where the error occurred is still indicated.
 ```
A command 'catkin build -i -v --summarize  --no-status  -p4 --make-args -j8' is running quiet mode. Elapsed: 1500
+++trap - EXIT
+++catkin_test_results build
moveit_planners_ompl/test_results/moveit_planners_ompl/MISSING-gtest-test_state_space.xml: 1 tests, 0 errors, 1 failures
Summary: 75 tests, 0 errors, 1 failures
+++error
+++travis_time_end 31
+++set +x
```

130s commented:
> Example of failing build in quiet mode (in [130s/moveit](https://travis-ci.org/130s/moveit/jobs/129664124#L4150). As you see where the error occurred is still indicated.

[The jobs there](https://travis-ci.org/130s/moveit/builds/129664121) are the example of too long jobs. Without this PR jobs easily overflow TravisCI's 4MB limit. With this PR, less than 5K lines.

gavanderhoorn commented:
I have very little time this week, but I can take a look later.

ipa-mdl commented:
I have tested it with `ros_canopen` and the output was reduced from 8539 to 3212 lines, great!
What do you think of making it the default and add a `VERBOSE_CATKIN_COMMANDS` flag instead?

130s commented:
>  What do you think of making it the default and add a VERBOSE_CATKIN_COMMANDS flag instead?

I'm inclined to that, but I'm just not sure yet; if reducing the result of build tool is ok, then the default output of tools (e.g. `make`, `gcc`, `catkin`) would have been much less. 
I'm just ignorant about these things so if we can make a consensus I'm happy to make "less output" option as the default.

Btw, @rhaschke made another PR https://github.com/ros-industrial/industrial_ci/pull/48 that sounds basically aiming the same. I like to get the best of both if that's ok.

130s commented:
I've changed my mind. Now "quiet" option is default and I added an option to enable verbosity. Thoughts were:

- On Travis CI, after 10,000 lines you're forced to see a plain text output, which is a pain without color and fold features. So shorter output is probably more beneficial than verbose output.
- You can always have a job to use verbosity. Maybe run 2 jobs (default and verbose) for the same config.

130s commented:
I might not be satisfied with what I've done so far; I'm [testing to run this PR on my personal repo (against industrial_core)](https://github.com/130s/industrial_core/pull/5) and no info is printed for the error for [one of the jobs failing](https://travis-ci.org/130s/industrial_core/jobs/134707097), which makes debugging almost impossible. I have to look.

130s commented:
There are other PRs and tickets (e.g. https://github.com/ros-industrial/industrial_ci/issues/55, https://github.com/ros-industrial/industrial_ci/pull/60) that could potentially change the length of the output logs, and I can't think of possible fix to [the issue I just raised in a prior comment](https://github.com/ros-industrial/industrial_ci/pull/43#issuecomment-223640058) (which I elaborated a little more below) for now. So I leave this PR as it is. In the future I may or may not come back to PR depending on how aforementioned tickets work out. 

- With the currently proposed change,

 https://travis-ci.org/130s/industrial_core/jobs/135741857#L3223-L3226

 ```
+++echo '[industrial_trajectory_filters] Started testing...'
[industrial_trajectory_filters] Started testing...
++++find install/share/industrial_trajectory_filters -iname '*.test'
find: `install/share/industrial_trajectory_filters': No such file or directory
```
 
 We can tell a cause of failure is related to `industrial_trajectory_filters` pkg but no info is available in the job log.

- Without the change, it's clear from [the log](https://travis-ci.org/ros-industrial/industrial_core/jobs/135489376#L2789-L2823) that `industrial_trajectory_filters` fails because [a depended package isn't available](https://github.com/ros-industrial/industrial_core/pull/144#issuecomment-223255184).

 ```
[industrial_trajectory_filters:cmake] CMake Error at /opt/ros/jade/share/catkin/cmake/catkinConfig.cmake:83 (find_package):
[industrial_trajectory_filters:cmake]   Could not find a package configuration file provided by
[industrial_trajectory_filters:cmake]   "moveit_ros_planning" with any of the following names:
[industrial_trajectory_filters:cmake] 
[industrial_trajectory_filters:cmake]     moveit_ros_planningConfig.cmake
[industrial_trajectory_filters:cmake]     moveit_ros_planning-config.cmake
[industrial_trajectory_filters:cmake] 
[industrial_trajectory_filters:cmake]   Add the installation prefix of "moveit_ros_planning" to CMAKE_PREFIX_PATH
[industrial_trajectory_filters:cmake]   or set "moveit_ros_planning_DIR" to a directory containing one of the above
[industrial_trajectory_filters:cmake]   files.  If "moveit_ros_planning" provides a separate development package or
[industrial_trajectory_filters:cmake]   SDK, be sure it has been installed.
[industrial_trajectory_filters:cmake] Call Stack (most recent call first):
[industrial_trajectory_filters:cmake]   CMakeLists.txt:5 (find_package)
[industrial_trajectory_filters:cmake] 
[industrial_trajectory_filters:cmake] 
[industrial_trajectory_filters:cmake] -- Could not find the required component 'moveit_ros_planning'. The following CMake error indicates that you either need to install the package with the same name or change your environment so that it can be found.
[industrial_trajectory_filters:cmake] -- Configuring incomplete, errors occurred!
```

davetcoleman commented on travis.sh:
```diff
@@ -233,8 +234,14 @@ source /opt/ros/$ROS_DISTRO/setup.bash # re-source setup.bash for setting enviro
-if [ "$BUILDER" == catkin ]; then catkin build -i -v --summarize  --no-status $BUILD_PKGS $CATKIN_PARALLEL_JOBS --make-args $ROS_PARALLEL_JOBS            ; fi
-
+if [ "$BUILDER" == catkin ]; then
+    _COMMAND_CATKINBUILD="catkin build -i -v --summarize  --no-status $BUILD_PKGS $CATKIN_PARALLEL_JOBS --make-args $ROS_PARALLEL_JOBS"
```
why do you use ``-v`` for your catkin commands?
 
>  --verbose, -v         Print output from commands in ordered blocks once the
>                        command finishes.

I don't use that flag locally in my development but I'm able to see enough feedback from catkin to fix bugs when they occur. I think that would save a lot of log space (and moveit doesn't have any to spare)

ipa-mdl commented:
I think this one can be closed

130s commented:
Superseded and realized by #94 and some more subsequent PRs.
